hydrolib/core/dflowfm/ini/util.py

```diff
@@ -622,3 +623,48 @@ def rename_keys_for_backwards_compatibility(
                 break
 
     return values
+
+
+class UnknownKeywordErrorManager:
+    """
+    Error manager for unknown keys.
+    Detects unknown keys and manages the Error to the user.
+    """
+
+    def raise_error_for_unknown_keywords(
+        self,
+        data: Dict[str, Any],
+        section_header: str,
+        fields: Dict[str, Any],
+        excluded_fields: Set,
+    ):
+        """
+        Notify the user of unknown keywords.
+
+        Args:
+            data (Dict[str, Any])   : Input data containing all set properties which are checked on unknown keywords.
+            section_header (str)    : Header of the section in which unknown keys might be detected.
+            fields (Dict[str, Any]) : Known fields of the section.
+            excluded_fields (Set)   : Fields which should be excluded from the check for unknown keywords.
+        """
+        unknown_keywords = self._get_all_unknown_keywords(data, fields, excluded_fields)
+
+        if len(unknown_keywords) == 0:
+            return
+        
+        raise ValueError(f"Unknown keywords are detected in section: '{section_header}', '{unknown_keywords}'")
+
+    def _get_all_unknown_keywords(
+        self, data: Dict[str, Any], fields: Dict[str, Any], excluded_fields: Set
+    ) -> List[str]:
+        list_of_unknown_keywords = []
+        for name, _ in data.items():
+            if self._is_unknown_keyword(name, fields, excluded_fields):
+                list_of_unknown_keywords.append(name)
+
+        return list_of_unknown_keywords
+
+    def _is_unknown_keyword(
+        self, name: str, fields: Dict[str, Any], excluded_fields: Set
+    ):
+        return name not in fields and name not in excluded_fields
```

Comment:
In HYDROLIB-core we use pydantic to turn dictionaries into valid objects. Pydantic can map the key in the dictionary to the appropriate attribute by either looking at the field name or the field alias. This behaviour can be set in the config (`config.allow_population_by_field_name`). In our `BaseModel` we set `config.allow_population_by_field_name` to `True`. This means that we want to be able to map the data via either the name or the alias.

In your implementation, we say that a keyword is unknown if there is no attribute with the same name as the key. However, this will, for example, fail for the research keywords that are implemented in #642. All research keyword attribute names are prefixed with _research_. This will fail the mapping and they will be considered an unknown keyword. I therefore think that it is important to not only check the field names, but also their alias!

Reply:
how would you suggest doing the alias check?
Create a set of all the aliases of the fields and check if the name is not in that set?